core/src/main/java/org/bf2/srs/fleetmanager/execution/impl/tasks/ScheduleRegistryTask.java

```diff
@@ -13,12 +15,15 @@
 @ToString(callSuper = true)
 public class ScheduleRegistryTask extends AbstractTask {
```

Comment:
The task data is de/serialized using JSON, and can be assigned to run on any node. We have to make sure that during upgrade, i.e. when some of the nodes were updated and others are running the old code, there is no deserialization conflict.

Reply:
De/serialization happens here `org.bf2.srs.fleetmanager.execution.manager.impl.JobWrapper#loadTask`

Reply:
I think the solution could be to introduce a new set of task types and workers or adding some logic into `org.bf2.srs.fleetmanager.execution.manager.Worker#supports` methods.

Reply:
Or maybe changing the way tasks are de/serialized `@JsonTypeInfo(use = JsonTypeInfo.Id.CLASS, property = "$class")` to make these changes more resilient.

Reply:
If we don't change the data of the task (as I'm planning with the next round of changes) then all of this is just not needed, correct?

Reply:
yes